plugins/PrivacyManager/Model/DataSubjects.php

```diff
@@ -352,6 +420,18 @@ private function visitsToWhereAndBind($tableToSelect, $visits)
         return array($where, $bind);
     }
 
+    private function getWhereToChooseVisitsForOtherSites($tableToSelect, $idSites)
+    {
+        $idSites = array_map('intval', $idSites);
+
+        // we also make sure we don't delete sites greater than the max idSite. this way if a site is added during
+        // an ongoing delete, the new valid data won't be deleted.
+        $maxIdSite = max($idSites);
+
+        $where = "$tableToSelect.idsite NOT IN (" . implode(',', $idSites) . ") AND $tableToSelect.idsite <= ?";
```

Comment:
it would probably be better here to JOIN to the site table so it works fast when there are 10,000 websites? as I think the NOT IN (10000 ids here) will not work well IIRC